GCRCatalogs/register.py

```diff
@@ -33,6 +76,21 @@ def load_yaml(yaml_file):
         config = yaml.safe_load(r.raw)
     return config
 
+def _resolve_dict(d):
+    for (k,v) in d.items():
+        if k in _PATH_LIKE_KEYS and isinstance(v, str):
+            if v.startswith('/'):
+                continue
+            elif _ROOT_DIR is None:
+                raise Exception('GCRCatalogs root dir not set')
+            else:
+                d[k] = os.path.join(_ROOT_DIR, v)
+        elif k in _DICT_LIST:
+            # list of dicts, each of which may have _PATH_LIKE keywords
+            for c in v:
+                _resolve_dict(c)
```

Comment:
I believe one of the test cases (the one with a composite catalog) shows that this does modify the input dictionary.  All that needs to be changed is the value of certain keywords.  It is not necessary to copy the whole dictionary.

Reply:
you're right. We should make it clear that `c` is a dictionary so that this is clear.

Reply:
I think it's clear as it stands.  There is a comment saying k is a list of dicts and the routine is called _resolve_dict.    I'd like to resolve this conversation.

Reply:
That's fine. 

Please add the checks (`isinstance(v, list) and all(isinstance(c, dict) for c in v)`) to avoid keyword collision (i.e., some other readers use `catalogs` for a different use).  Thanks,